src/Compilers/CSharp/Portable/Binder/LocalBinderFactory.cs

```diff
@@ -206,6 +208,45 @@ public override void VisitArrowExpressionClause(ArrowExpressionClauseSyntax node
             Visit(node.Expression, arrowBinder);
         }
 
+        public override void VisitEqualsValueClause(EqualsValueClauseSyntax node)
+        {
+            var valueBinder = new PatternVariableBinder(node, node.Value, _enclosing);
+            AddToMap(node, valueBinder);
+            Visit(node.Value, valueBinder);
+        }
+
+        public override void VisitAttribute(AttributeSyntax node)
+        {
+            var attrBinder = new PatternVariableBinder(node, _enclosing);
+            AddToMap(node, attrBinder);
+
+            if (node.ArgumentList?.Arguments.Count > 0)
+            {
+                foreach (var argument in node.ArgumentList.Arguments)
+                {
+                    Visit(argument.Expression, attrBinder);
+                }
+            }
+        }
+
+        public override void VisitArgumentList(ArgumentListSyntax node)
```

Comment:
I don't understand this. An argument list doesn't define a separate scope, does it?